server/internal/core/application/covenantless.go

```diff
@@ -1469,68 +1481,92 @@ func (s *covenantlessService) finalizeRound(notes []note.Note, roundEndTime time
 		}
 	}()
 
-	remainingTime := time.Until(roundEndTime)
-	select {
-	case <-s.forfeitsBoardingSigsChan:
-		log.Debug("all forfeit txs and boarding inputs signatures have been sent")
-	case <-time.After(remainingTime):
-		log.Debug("timeout waiting for forfeit txs and boarding inputs signatures")
-	}
-
-	forfeitTxs, err := s.forfeitTxs.pop()
-	if err != nil {
-		changes = round.Fail(fmt.Errorf("failed to finalize round: %s", err))
-		log.WithError(err).Warn("failed to finalize round")
-		return
-	}
-
-	if err := s.verifyForfeitTxsSigs(forfeitTxs); err != nil {
-		changes = round.Fail(err)
-		log.WithError(err).Warn("failed to validate forfeit txs")
-		return
-	}
-
-	log.Debugf("signing round transaction %s\n", round.Id)
-
-	boardingInputsIndexes := make([]int, 0)
-	boardingInputs := make([]domain.VtxoKey, 0)
 	roundTx, err := psbt.NewFromRawBytes(strings.NewReader(round.UnsignedTx), true)
 	if err != nil {
 		log.Debugf("failed to parse round tx: %s", round.UnsignedTx)
 		changes = round.Fail(fmt.Errorf("failed to parse round tx: %s", err))
 		log.WithError(err).Warn("failed to parse round tx")
 		return
 	}
+	includesBoardingInputs := false
+	for _, in := range roundTx.Inputs {
+		scriptType := txscript.GetScriptClass(in.WitnessUtxo.PkScript)
+		if scriptType == txscript.WitnessV1TaprootTy {
```

Comment:
nit: this is not enough if the server is using taproot utxos in the future, maybe worth to add a comment?